src/SWP/Bundle/ContentBundle/Resolver/AssetLocationResolver.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2019 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2019 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\ContentBundle\Resolver;
+
+use Aws\S3\S3Client;
+use SWP\Bundle\ContentBundle\Model\FileInterface;
+
+class AssetLocationResolver implements AssetLocationResolverInterface
+{
+    private $mainAdapter;
+    private $awsBucket;
+    private $awsClient;
+    private $localDirectory;
+
+    public function __construct(string $mainAdapter, string $awsBucket, S3Client $awsClient, string $localDirectory)
+    {
+        $this->mainAdapter = $mainAdapter;
+        $this->awsBucket = $awsBucket;
+        $this->localDirectory = $localDirectory;
+        $this->awsClient = $awsClient;
+    }
+
+    public function getAssetUrl(FileInterface $file): string
+    {
+        if ('aws_adapter' === $this->mainAdapter) {
```

Comment:
you can put that into a separate class `AssetUrlGenerator` with a separate interface and one method `getUrl` and inject that service here.

Then it should be possible to register custom generator, for example, AwsUrlGenerator, LocalUrlGenerator (by adding tags `swp.asset_url_generator` to it) and in the future, if you would want to use another storage you could easily add a new class to support it without making changes to the existing code. Because right now it's all hardcoded here.